tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ManagedIdentityTests.cs

```diff
@@ -0,0 +1,325 @@
+            {
+                exception = ex.Message;
+            }
+
+            //Assert
+            Assert.IsTrue(string.IsNullOrEmpty(exception));
+            Assert.AreSame(result, expectedClientException);
+        }
+
+        [DataTestMethod]
+        [DataRow(MsiAzureResource.WebApp, "", DisplayName = "System Identity Web App")]
+        [DataRow(MsiAzureResource.Function, "", DisplayName = "System Identity Function App")]
+        [DataRow(MsiAzureResource.WebApp, UserAssignedClientID, DisplayName = "User Identity Web App")]
+        [DataRow(MsiAzureResource.Function, UserAssignedClientID, DisplayName = "User Identity Function App")]
+        [DataRow(MsiAzureResource.WebApp, Mi_res_id, DisplayName = "ResourceID Web App")]
+        [DataRow(MsiAzureResource.Function, Mi_res_id, DisplayName = "ResourceID Function App")]
+        public async Task AcquireMSITokenAsync(MsiAzureResource azureResource, string userIdentity)
+        {
+            //Arrange
+            AuthenticationResult result = null;
+
+            //Set the Environment Variables
+            var environmentVariables = await GetEnvironmentVariablesAsync(azureResource)
+                .ConfigureAwait(false);
+
+            SetEnvironmentVariables(environmentVariables);
+
+            //form the URI 
+            string uri = s_baseURL + $"GetMSIToken?azureresource={azureResource}&uri=";
+
+            //Create CCA with Proxy
+            IConfidentialClientApplication cca = CreateCCAWithProxy(uri);
+
+            //Act
+            try
+            {
+                result = await cca.AcquireTokenForClient(s_msi_scopes)
+                    .WithManagedIdentity(userIdentity)
+                    .ExecuteAsync().ConfigureAwait(false);
+            }
+            catch (MsalClientException ex)
+            {
+                throw new Exception(ex.Message);
+            }
+
+            //Assert
+            Assert.IsNotNull(result.AccessToken);
+
+            CoreAssert.IsWithinRange(
+                            DateTimeOffset.UtcNow + TimeSpan.FromHours(0),
+                            result.ExpiresOn,
+                            TimeSpan.FromHours(24));
+
+            Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+
+            result = await cca.AcquireTokenForClient(s_msi_scopes)
+                .WithManagedIdentity(userIdentity)
+                .ExecuteAsync().ConfigureAwait(false);
+
+            Assert.IsNotNull(result.Scopes);
+            Assert.AreEqual(TokenSource.Cache, result.AuthenticationResultMetadata.TokenSource);
+
+            ClearEnvironmentVariables(environmentVariables);
+
+        }
+
+        [DataTestMethod]
+        [DataRow(MsiAzureResource.WebApp, "", SendMSIHeader.None, DisplayName = "Web App No Header")]
+        [DataRow(MsiAzureResource.WebApp, "", SendMSIHeader.WithWrongValue, DisplayName = "Web App Wrong Header")]
+        public async Task TryAcquireMSITokenHeaderTestAsync(
```

Comment:
@bgavrilMS @neha-bhargava please take a look at these tests, most of these may not be applicable since I am tweaking the request before it is sent out. But find a few cases where the error message is missing a co-correlation ID or cases where we do not throw the MSI endpoint error message.

Reply:
I disabled these as they may not be useful for integration, but @neha-bhargava just the https://portal.azure.com token did not work, maybe you can see why?

Reply:
When would we send MSI header with none or wrong value?

Reply:
we get the value and then the value changes in the web app. Web restarts or MSI endpoint updates can change the header value. We can discuss more when we write the negative cases

Reply:
Should we tweak the request sent out? Will this ever happen?

Reply:
Yes, lets discuss these.